packages/block-editor/src/components/block-popover/index.js

```diff
@@ -88,6 +93,20 @@ function BlockPopover(
 
 		return {
 			getBoundingClientRect() {
+				if ( isZoomOut() ) {
```

Comment:
I think this would benefit from a comment to explain why this additional logic is necessary in Zoom Out. What specifically necessitates this? It will help avoid a future developer needing to gain a full understanding of the specifics of Zoom Out mode UI.